packages/forms/src/UIForm/fields/ListView/ListView.component.js

```diff
@@ -0,0 +1,222 @@
+
+import { getItemsProps, initItems, updateItems } from './ListView.utils';
+import FieldTemplate from '../FieldTemplate';
+
+const DISPLAY_MODE_DEFAULT = 'DISPLAY_MODE_DEFAULT';
+const DISPLAY_MODE_SEARCH = 'DISPLAY_MODE_SEARCH';
+const DEFAULT_ITEM_HEIGHT = 33;
+
+class ListViewWidget extends React.Component {
+	constructor(props) {
+		super(props);
+
+		this.defaultHeaderActions = [
+			{
+				icon: 'talend-search',
+				id: `${props.id}-search`,
+				label: 'Search for specific values',
+				onClick: this.switchToSearchMode.bind(this),
+			},
+		];
+		this.searchHeaderActions = [
+			{
+				label: 'Abort',
+				icon: 'talend-cross',
+				id: 'abort',
+				onClick: this.switchToDefaultMode.bind(this),
+			},
+		];
+
+		this.state = {
+			...initItems(props.schema, props.value, null, this.onToggleItem.bind(this)),
+			getItemHeight: () => DEFAULT_ITEM_HEIGHT,
+			headerDefault: this.defaultHeaderActions,
+			onAddKeyDown: this.onInputKeyDown.bind(this),
+			onInputChange: this.onInputChange.bind(this),
+			onToggleAll: this.onToggleAll.bind(this),
+		};
+	}
+
+	/**
+	 * On new schema : we redefine the items
+	 * On new value : we update the check status
+	 * @param { Object } schema The new mergedSchema
+	 * @param { Array } value The new value
+	 */
+	componentWillReceiveProps({ schema, value }) {
+		if (schema !== this.props.schema) {
+			this.setState(oldState =>
+				initItems(schema, value, oldState.searchCriteria, this.onToggleItem.bind(this)),
+			);
+		} else if (value !== this.props.value) {
+			this.setState(oldState =>
+				updateItems(oldState.items, value, oldState.searchCriteria)
+			);
+		}
+	}
+
+	/**
+	 * Propagate a ListView value change
+	 * @param { Object } event The event that triggered the change
+	 * @param { Array } newValue The new Value
+	 */
+	onChange(event, newValue) {
+		const value = newValue.length ? newValue : undefined;
+		const payload = { schema: this.props.schema, value };
+		this.props.onChange(event, payload);
+		this.props.onFinish(event, payload);
+	}
+
+	/**
+	 * Search input change
+	 * @param { Object } event The change event
+	 * @param { Array } value
+	 */
+	onInputChange(event, { value }) {
+		if (this.timerSearch) {
+			clearTimeout(this.timerSearch);
+		}
+		this.timerSearch = setTimeout(() => {
+			this.setState(getItemsProps(this.state.items, value));
+			this.timerSearch = null;
+		}, 400);
+	}
+
+	/**
+	 * Search input ENTER/ESC management
+	 * @param { Object } event The keydown event
+	 */
+	onInputKeyDown(event) {
+		if (event.keyCode === keycode('enter')) {
+			event.stopPropagation();
+			event.preventDefault();
+		}
+		if (event.keyCode === keycode('escape')) {
+			event.stopPropagation();
+			event.preventDefault();
+			this.switchToDefaultMode();
+		}
+	}
+
+	/**
+	 * Toggle an item
+	 * @param { Object } event The toggle event
+	 * @param { Object } changedItem The item to toggle
+	 */
+	onToggleItem(event, changedItem) {
+		const value = this.state.items
+			.filter((item) => {
```

Comment:
if im not mistaken only element is affected each time, a find will be faster in this case, since its stop iterrating once the value is found.

Reply:
No this rebuild the new value, so we have to iterate over all values
It's just that if we toggle item X, we need to test the reverse checked value to know if it is in the values